src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -53,7 +57,7 @@ internal sealed partial class MiscellaneousFilesWorkspace : Workspace, IVsRunnin
             SaveEventsService saveEventsService,
             VisualStudioWorkspace visualStudioWorkspace,
             SVsServiceProvider serviceProvider) :
-            base(visualStudioWorkspace.Services.HostServices, WorkspaceKind.MiscellaneousFiles)
+            base(new CustomizedHostServices(visualStudioWorkspace.Services.HostServices, GetWorkspaceSpecificService), WorkspaceKind.MiscellaneousFiles)
```

Comment:
Why not just export an `IScriptEnvironmentService` with [`WorkspaceKind.MiscellaneousFiles`](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/WorkspaceKind.cs,47d3c0cca9e894bc)? That's how workspace services are intended to be registered for MEF hosts.